### PR TITLE
ENH explicit checks and PR comments

### PR DESCRIPTION
**Pull request by beckermr**
OK @CJ-Wright here is the PR. I still need to test it live.

It has
- explicit checks
- comments on PRs to tell people what happened

### CODE REVIEW

File: conda_forge_tick_action/automerge.py [hunk 1]
```diff
@@ -26,17 +26,40 @@
...
+
+def _get_conda_forge_config(pr):
+    """get the conda-forge.yml from upstream master
+
+    We always do this to make sure we use the maintainer settings and not
+    any from a fork.
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        subprocess.run(
+            ["git", "clone", pr.base.repo.clone_url, tmpdir],
```
Comment (isuruf) [inline conda_forge_tick_action/automerge.py]: Shouldn't this use the target branch as well?

Comment (beckermr) [reply conda_forge_tick_action/automerge.py]: Ah the target branch of the pr. Yes it should.

File: conda_forge_tick_action/automerge.py [hunk 2]
```diff
@@ -139,16 +158,111 @@ def _check_github_statuses(statuses, extra_ignored_statuses=None):
...
+        r.lower()
+        for r in required
+        if not any(r.lower() in _i for _i in ignored_statuses)
+    ]
+
+
+def _all_statuses_and_checks_ok(
+    status_states, check_states, req_checks_and_states
+):
+    """check all of the required statuses are OK"""
```
Comment (beckermr) [inline conda_forge_tick_action/automerge.py]: Doc string is wrong. 

Also this needs to return none for pending things.

**Review by beckermr — COMMENTED** [review]
The bot should comment only if all final Statuses are not none.

File: conda_forge_tick_action/automerge.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick_action/automerge.py]: done.

File: conda_forge_tick_action/automerge.py (hunk 2 above)
Comment (beckermr) [reply conda_forge_tick_action/automerge.py]: done.

**Comment by beckermr** [comment]
OK @CJ-Wright @isuruf this PR is ready for another look!

I have tested it live using the scripts in the scripts directory. I put them in this repo so that others can use them later.

**Review by CJ-Wright — APPROVED** [review]
Seems reasonable, although I had difficulty reading through it all

**Pull request merged**